gen/calc.py

```diff
@@ -15,6 +15,21 @@
 from pkgpanda.build import hash_checkout
 
 
+def validate_one_of(val, valid_values) -> None:
+    """Test if object `val` is a member of container `valid_values`.
+    Raise a ValueError if it is not a member. The exception message contains
+    both, the representation (__repr__) of `val` as well as the representation
+    of all items in `valid_values`.
+    """
+    if val not in valid_values:
+        options_string = ', '.join("'{}'".format(v) for v in valid_values)
```

Comment:
Let's make sure that `__repr__()` builds the string instead of `__str__()`,  via `{0!r}` (or `%r` in classical string formatting). That gives most insightful error messages.